core/server/master/src/main/java/alluxio/underfs/MasterUfsManager.java

```diff
@@ -11,17 +11,86 @@
 
 package alluxio.underfs;
 
+import alluxio.exception.InvalidPathException;
+import alluxio.util.UnderFileSystemUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
  * A class that manages the UFS for master servers.
  */
 @ThreadSafe
 public final class MasterUfsManager extends AbstractUfsManager {
+  private static final Logger LOG = LoggerFactory.getLogger(MasterUfsManager.class);
+
+  // The physical ufs state for all active mounts if not the default
+  private ConcurrentHashMap<String, UnderFileSystem.UfsMode> mPhysicalUfsState =
+      new ConcurrentHashMap<>();
```

Comment:
Whats the cost of being explicit and including everything?

Reply:
Primarily to prevent user error. Using the CLI w/o knowledge of physical  ufs, never validates a ufs as something used otherwise. The secondary reason is unbounded state on the master which is never garbage collected (less important given the size).